Scenario3/PetImagesTest/StorageMocks/MockCosmosContainer.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.Coyote.Random;
+using PetImages.Entities;
+using PetImages.Storage;
+using PetImagesTest.Exceptions;
+
+namespace PetImagesTest.StorageMocks
+{
+    public class MockCosmosContainer : ICosmosContainer
+    {
+        private readonly string ContainerName;
+        private readonly MockCosmosState State;
+        private readonly Generator Generator;
+        private bool EmitRandomizedFaults;
+
+        public MockCosmosContainer(string containerName, MockCosmosState state)
+        {
+            this.ContainerName = containerName;
+            this.State = state;
+            this.Generator = Generator.Create();
+            this.EmitRandomizedFaults = false;
+        }
+
+        public Task<T> CreateItem<T>(T item)
+            where T : DbItem
+        {
+            var itemCopy = TestHelper.Clone(item);
+
+            return Task.Run(() =>
+            {
+                if (this.EmitRandomizedFaults && this.Generator.NextBoolean())
+                {
+                    throw new SimulatedDatabaseFaultException();
+                }
+
+                this.State.CreateItem(this.ContainerName, itemCopy);
```

Comment:
@imnaseer Do we need to bug up these mocks? since these mocks will have correct versions in the scenario 1. Hence kept them as is